### PR TITLE
:sparkles: cli: added export to json on get_signatures

### PR DESCRIPTION
**Pull request by callmepak**

### CODE REVIEW

**Comment by callmepak** [comment]
#LSC-215

File: cmd/dc4bc_cli/main.go [hunk 1]
```diff
@@ -260,6 +268,30 @@ func getSignaturesCommand() *cobra.Command {
 					fmt.Println()
 				}
 			}
+
+			if jsonOutputFolder == "" {
+				return nil
+			}
+
+			f, err := os.OpenFile(jsonOutputFolder, os.O_WRONLY|os.O_CREATE, 0600)
```
Comment (sotnikov-s) [inline cmd/dc4bc_cli/main.go:276]: why do you open it if it's a folder? 🤔 
btw, why is it a folder at all? according to the task, it should be a json file with an array of signatures

File: cmd/dc4bc_cli/main.go [hunk 2]
```diff
@@ -260,6 +267,32 @@ func getSignaturesCommand() *cobra.Command {
 					fmt.Println()
 				}
 			}
+
+			if jsonOutputFolder == "" {
+				return nil
+			}
+
+			filename := fmt.Sprintf("%s.json", dkgID)
```
Comment (swelf19) [inline cmd/dc4bc_cli/main.go:275]: i would suggest request-style name, e.g. `dkg_signatures_dump_<short_id>.json`

File: HowTo.md [hunk 3]
```diff
@@ -327,6 +327,15 @@ Signing ID: ca800cac-2c13-4a14-8ca3-72c36112c5e4
                 Reconstructed signature for the data:
 ```
 
+To save this list of all messages to json, you should use this command with flag --json_files_folder:
+```
+./dc4bc_cli get_signatures c04f3d54718dfc801d1cbe86e3a265f5342ec2550f82c1c3152c36763af3b8f2 --json_files_folder .
```
Comment (swelf19) [inline HowTo.md:332]: according to the init function in dc4bc_cli default value for `json_files_folder` is `/tmp`, and signatures always saves to the output dir. Comment looks a little bit misleading

File: HowTo.md [hunk 4]
```diff
@@ -320,11 +322,24 @@ But before signing, spend some time on taking a look at what you're about to sig
 
 At this point the other participants would probably like to take a look at the message they've been proposed to sign. To do so, they can run the following command that reveals a list of all messages related to a given DKG round as well as the messages signing IDs and hashes.
 ```
-./dc4bc_cli get_signatures c04f3d54718dfc801d1cbe86e3a265f5342ec2550f82c1c3152c36763af3b8f2
-Signing ID: ca800cac-2c13-4a14-8ca3-72c36112c5e4
-                DKG round ID: c04f3d54718dfc801d1cbe86e3a265f5342ec2550f82c1c3152c36763af3b8f2
-                Participant: john_doe
-                Reconstructed signature for the data:
+./dc4bc_cli get_signatures c04f3d54718dfc801d1cbe86e3a265f5342ec2550f82c1c3152c36763af3b8f2 --json_files_folder /tmp/
```
Comment (pr0n00gler) [inline HowTo.md:325]: `export_signatures`?

File: cmd/dc4bc_cli/main.go [hunk 5]
```diff
@@ -229,37 +232,89 @@ func getSignaturesRequest(host string, dkgID string) (*SignaturesResponse, error
...
 			}
-			for sigID, signature := range signatures.Result {
-				fmt.Printf("Signing ID: %s\n", sigID)
-				for _, participantSig := range signature {
-					fmt.Printf("\tDKG round ID: %s\n", participantSig.DKGRoundID)
-					fmt.Printf("\tParticipant: %s\n", participantSig.Username)
-					fmt.Printf("\tReconstructed signature for the data: %s\n", base64.StdEncoding.EncodeToString(participantSig.Signature))
-					fmt.Println()
+
+			if printFullSignaturesInfo {
```
[the remark this answers is not in the document]
Comment (oopcode) [reply cmd/dc4bc_cli/main.go:271]: In general you are right, but in our use case there is going to be thousands of signatures times a dozen of signers, which renders this optional print functionality basically useless. I would rather opt to remove it at all, tbh, but it's good for debugging purposes. 

So, to conclude: your point is valid, but given the circumstances I don't see enough reasons to make a separate command for printing.

File: HowTo.md [hunk 6]
```diff
@@ -162,14 +162,14 @@ Example of start_dkg_propose.json file structure:
   "SigningThreshold": 2,
   "Participants": [
     {
-      "Username": "john_doe",
-      "PubKey": "EcVs+nTi4iFERVeBHUPePDmvknBx95co7csKj0sZNuo=",
-      "DkgPubKey": "sN7XbnvZCRtg650dVCCpPK/hQ/rMTSlxrdnvzJ75zV4W/Uzk9suvjNPtyRt7PDXLDTGNimn+4X/FcJj2K6vDdgqOrr9BHwMqJXnQykcv3IV0ggIUjpMMgdbQ+0iSseyq"
+      "Username": "user1",
```
[the remark this answers is not in the document]
Comment (oopcode) [reply HowTo.md:165]: I just noticed that the public keys format was wrong, don't have anything against the family

image

File: cmd/dc4bc_cli/main.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (sotnikov-s) [reply cmd/dc4bc_cli/main.go:271]: please take a look at what I've come up with:
export_signatures with no flag provided: exports signatures in a json file
export_signatures with `--print-only` flag provided: only prints signatures, no json is created

take a look and think about it one more time. but if you will still think it's okay to leave it the way you've done it -- it's really totally fine. just want to make sure you've got my suggestion right

**Review by sotnikov-s — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Pull request merged**